### PR TITLE
[ebay.com] Clean-up ruleset

### PR DESCRIPTION
**Pull request by youdly**
Remove country domains for https://github.com/EFForg/https-everywhere/issues/13603
There are >3k subdomains listed by Sublist3r, which is impractical to test.
I rewrite the ruleset based on the subdomains listed in the comment of previous ruleset, which I assume are the most popular/relevant subdomains.

### CODE REVIEW

**Comment by youdly** [comment]
Removed ebay reference from (unrelated) rulesets.

**Comment by youdly** [comment]
Leave ebay reference in rulesets `CSdata1.com.xml` `ebayclassifieds.com.xml` `Gittigidiyor.xml` `gumtree.com.xml` `mobile.de.xml` `Shopping.com.xml` `Shoppingshadow.com.xml` untouched. Will clean them up in separate PRs.

**Comment by youdly** [comment]
After reading through https://github.com/EFForg/https-everywhere/issues/13762, I think I should be more careful around the `securecookie` tag. It seems ebay has yet to secure all subdomains yet, so adding the tag is a bit risky.

File: src/chrome/content/rules/ebay.com.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+<!--
+
+	Other eBay rulesets:
+
+		- ebay.at
+		- ebay.be
+		- ebay.ca
+		- ebay.ch
+		- ebay.co.uk
+		- ebay.com
```
Comment (J0WI) [inline src/chrome/content/rules/ebay.com.xml]: Please remove the reference to itself and add `.xml` to all of them.
Not all ccTLD rulesets do exist. Do you plan to add them all before we merge this?

The previous references need to be updated too:
```
- CSdata1.com.xml
- ebayclassifieds.com.xml
- Gittigidiyor.xml
- gumtree.com.xml
- Shopping.com.xml
- Shoppingshadow.com.xml
- mobile.de.xml
```

Comment (youdly) [reply src/chrome/content/rules/ebay.com.xml]: No, currently only those ccTLD. I'll just add `ebayclassifiedsgroup.com.xml`.
https://github.com/EFForg/https-everywhere/issues/13603#issuecomment-348856123

**Comment by youdly** [comment]
Added redirect `ebay.com` -> `www.ebay.com`.

**Comment by youdly** [comment]
Temporarily remove dangling references `ebay.co.uk.xml`, `ebay.com.sg.xml`, `ebayrtm.com.xml`.

**Comment by youdly** [comment]
rebased

**Comment by youdly** [comment]
In addition to https://github.com/EFForg/https-everywhere/pull/13773#issuecomment-355730754, `ebayclassifiedsgroup.com.xml` is also removed.

**Comment by youdly** [comment]
Remove `ebaydesc.com.xml` reference (host is non-functional). Squashed some commit.

**Comment by Bisaloo** [comment]
Related: https://github.com/EFForg/https-everywhere/issues/11203

**Comment by youdly** [comment]
Currently this PR only depends on `ebaycommercenetwork.com.xml` https://github.com/EFForg/https-everywhere/pull/14259

File: src/chrome/content/rules/ebay.com.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	<target host="reco.ebay.com" />
+	<target host="rover.ebay.com" />
+	<target host="scgi.ebay.com" />
+	<target host="securethumbs.ebay.com" />
+	<target host="signin.ebay.com" />
+	<target host="spages.half.ebay.com" />
+	<target host="srtm.ebay.com" />
+	<target host="svcs.ebay.com" />
+	<target host="thumbs.ebay.com" />
+	<target host="viv.ebay.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/ebay.com.xml]:
```
anywhere.ebay.com
charity.ebay.com
developer.ebay.com
forums.developer.ebay.com
go.developer.ebay.com
donation.ebay.com
explore.ebay.com
givingworks.ebay.com
ocsnext.ebay.com
partnernetwork.ebay.com
resolutioncenter.ebay.com
sellforme.ebay.com
shopbot.ebay.com

Comment (youdly) [reply src/chrome/content/rules/ebay.com.xml]: Added

**Comment by J0WI** [comment]
Thanks @youdly!

**Pull request merged**